pkg/reconciler/pipelinerun/resources/pipelineref.go

```diff
@@ -86,24 +87,54 @@ func GetPipelineFunc(ctx context.Context, k8s kubernetes.Interface, tekton clien
 			}
 			replacedParams := replaceParamValues(pr.Params, stringReplacements, arrayReplacements, objectReplacements)
 			resolver := resolution.NewResolver(requester, pipelineRun, string(pr.Resolver), "", "", replacedParams)
-			return resolvePipeline(ctx, resolver, name, k8s)
+			return resolvePipeline(ctx, resolver, name)
 		}
 	default:
 		// Even if there is no task ref, we should try to return a local resolver.
 		local := &LocalPipelineRefResolver{
 			Namespace:    namespace,
 			Tektonclient: tekton,
-			K8sclient:    k8s,
 		}
 		return local.GetPipeline
 	}
 }
 
+// GetVerifiedPipelineFunc is a wrapper of GetPipelineFunc and return the function to
+// verify the pipeline if resource-verification-mode is not "skip"
+func GetVerifiedPipelineFunc(ctx context.Context, k8s kubernetes.Interface, tekton clientset.Interface, requester remoteresource.Requester, pipelineRun *v1beta1.PipelineRun, verificationpolicies []*v1alpha1.VerificationPolicy) rprp.GetPipeline {
+	get := GetPipelineFunc(ctx, k8s, tekton, requester, pipelineRun)
+	return func(context.Context, string) (v1beta1.PipelineObject, *v1beta1.ConfigSource, error) {
+		p, s, err := get(ctx, pipelineRun.Spec.PipelineRef.Name)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to get pipeline: %w", err)
+		}
+		// if the pipeline is in status, then it has been verified and no need to verify again
+		if pipelineRun.Status.PipelineSpec != nil {
+			return p, s, nil
+		}
+		var source string
+		if s != nil {
+			source = s.URI
+		}
+		logger := logging.FromContext(ctx)
+		if config.CheckEnforceResourceVerificationMode(ctx) || config.CheckCheckEnforceResourceVerificationMode(ctx){
+			if err := trustedresources.VerifyPipeline(ctx, p, k8s, source, verificationpolicies); err != nil {
+				if config.CheckEnforceResourceVerificationMode(ctx) {
+					logger.Errorf("GetVerifiedPipelineFunc failed: %v", err)
+					return nil, nil, fmt.Errorf("GetVerifiedPipelineFunc failed: %w: %v", trustedresources.ErrorResourceVerificationFailed, err)
+				}
+				logger.Warnf("GetVerifiedPipelineFunc failed: %v", err)
+				return p, s, nil
+			}
+		}
+		return p, s, nil
```

Comment:
previous `verifyResolvedPipeline` is refactored here to meet the 2 suggestions:

1. check feature flag in `GetVerifiedPipelineFunc`
2. doesn't contain `verify` function in this pkg

The changes are the same in `taskref.go`